evm_loader/program/src/evm/mod.rs

```diff
@@ -24,45 +25,48 @@ mod buffer;
 pub mod database;
 mod memory;
 mod opcode;
-mod opcode_table;
+#[cfg(feature = "library")]
+mod opcode_async;
+#[cfg(not(feature = "library"))]
+mod opcode_sync;
```

Comment:
You have two different ways to call opcodes. 
In `opcode_async`, you use `match` because not every opcode is async. Would it be possible to make all of them async and share the implementation between sync and async versions?

Reply:
I defined a declarative macro to remove some code duplication.

However, defining a method table for the async code is not possible without making the opcode methods return a [`LocalBoxFuture`](https://docs.rs/futures/latest/futures/future/type.LocalBoxFuture.html), because each opcode method returns a different opaque future type and all types need to be the same for all match arms in order to put them into a method table.